### PR TITLE
Make service principal a user input rather than creating on each depl…

### PR DESCRIPTION
**Pull request by persiaAziz**
…oyment

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,7 +49,10 @@ Example terraform.tfvars:
...
  install_xsa =  
+ service_principal_id = 
+ service_principal_password = 
+
 
 Additional steps
 -------------------------------------
-Please download the relevant SAP installers and store them in accessible blobs, and use SAS urls to those blobs for installation.
\ No newline at end of file
+Please download the relevant SAP installers and store them in accessible blobs, and use SAS urls to those blobs for installation.
```
Comment (pabowers) [inline README.md]: Tobias updated the documentation in the most recent version of master. Please be consistent with that.

File: README.md [hunk 2]
```diff
@@ -49,7 +49,10 @@ Example terraform.tfvars:
  pw_db_system =  
  useHana2 =  
  install_xsa =  
+ service_principal_id = 
+ service_principal_password = 
+
```
Comment (tniek) [inline README.md]: We should keep the nomenclature more consistent; e.g. all passwords start with "pw_" (see line 45 -> "pw_db_system").

File: README.md (hunk 1 above)
Comment (tniek) [reply README.md]: Maybe link to downloads.md for the time being (at least until we have a script that downloads the bits)?

File: deploy/vm/ansible/roles/stonith-device-creation/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+---
+spn_id:
+spn_pw:
```
Comment (tniek) [inline deploy/vm/ansible/roles/stonith-device-creation/defaults/main.yml]: spn = service principal? Not consistent with the above naming

File: deploy/vm/modules/ha_pair/main.tf [hunk 4]
```diff
@@ -182,4 +182,6 @@ module "configure_vm" {
   install_xsa                    = "${var.install_xsa}"
   install_shine                  = "${var.install_shine}"
   install_cockpit                = "${var.install_cockpit}"
+  service_principal_id           = "${var.service_principal_id}"
+  service_principal_password     = "${var.service_principal_password}"
```
Comment (tniek) [inline deploy/vm/modules/ha_pair/main.tf]: Naming (see above); just from the variable name, it's hard to tell which category it belongs to (OS/HANA/Azure/...)

File: deploy/vm/ansible/roles/stonith-device-creation/defaults/main.yml (hunk 3 above)
Comment (persiaAziz) [reply deploy/vm/ansible/roles/stonith-device-creation/defaults/main.yml]: yes.. spn is service principal. This is not a customer facing variable, so I went cheap on the naming. I can change it now.

File: README.md (hunk 2 above)
Comment (persiaAziz) [reply README.md]: done

File: deploy/vm/ansible/roles/stonith-device-creation/defaults/main.yml (hunk 3 above)
Comment (pabowers) [reply deploy/vm/ansible/roles/stonith-device-creation/defaults/main.yml]: I agree with Tobias with the naming

**Pull request merged**